org.eclipse.jgit.test/tst/org/eclipse/jgit/diff/DiffFormatterTest.java

```diff
@@ -343,6 +344,190 @@ public void testDiff() throws Exception {
+						+ "new file mode 100644\n"
+						+ "index 0000000..8013df8\n"
+						+ "--- /dev/null\n"
+						+ "+++ b/folder/folder.txt\n"
+						+ "@@ -0,0 +1 @@\n"
+						+ "+change\n"
+						+ "\\ No newline at end of file\n";
+
+		assertEquals(expected, actual);
+	}
+
+	@Test
+	/**
+	 * This is an DELETE file, the file content matches the filter: diff unchanged.
+	 */
+	public void testDiffDeltaFilter_deleteFile() throws Exception {
+		write(new File(db.getDirectory().getParent(), "test.txt"), "test");
+		File folder = new File(db.getDirectory().getParent(), "folder");
+		FileUtils.mkdir(folder);
+		write(new File(folder, "folder.txt"), "change");
+		Git git = new Git(db);
+		git.add().addFilepattern(".").call();
+		git.commit().setMessage("Initial commit").call();
+		new File(folder, "folder.txt").delete();
+
+		ByteArrayOutputStream os = new ByteArrayOutputStream();
+		DiffFormatter dfmt = new DiffFormatter(new SafeBufferedOutputStream(os));
+		dfmt.setRepository(db);
+		dfmt.setPathFilter(PathFilter.create("folder"));
+		DirCacheIterator oldTree = new DirCacheIterator(db.readDirCache());
+		FileTreeIterator newTree = new FileTreeIterator(db);
+
+		//testing a delta filter with one regex
+		Pattern deltaFilterPattern = Pattern.compile("change");
+		dfmt.format(dfmt.scan(oldTree, newTree), deltaFilterPattern);
+		dfmt.flush();
+
+		String actual = os.toString("UTF-8");
+		String expected =
+				"diff --git a/folder/folder.txt b/folder/folder.txt\n"
+						+ "deleted file mode 100644\n"
+						+ "index 8013df8..0000000\n"
+						+ "--- a/folder/folder.txt\n"
+						+ "+++ /dev/null\n"
+						+ "@@ -1 +0,0 @@\n"
+						+ "-change\n"
+						+ "\\ No newline at end of file\n";
+
+		assertEquals(expected, actual);
+	}
+
+	@Test
+	/**
+	 * Filter for any file matches the content of the changed file: diff skipped.
+	 */
+	public void testDiffDeltaFilter_filteredModifiedFile() throws Exception {
+		write(new File(db.getDirectory().getParent(), "test.txt"), "test");
+		File folder = new File(db.getDirectory().getParent(), "folder");
+		FileUtils.mkdir(folder);
+		write(new File(folder, "folder.txt"), "folder");
+		Git git = new Git(db);
+		git.add().addFilepattern(".").call();
+		git.commit().setMessage("Initial commit").call();
+		write(new File(folder, "folder.txt"), "folderchange");
+
+		ByteArrayOutputStream os = new ByteArrayOutputStream();
+		DiffFormatter dfmt = new DiffFormatter(new SafeBufferedOutputStream(os));
+		dfmt.setRepository(db);
+		dfmt.setPathFilter(PathFilter.create("folder"));
+		DirCacheIterator oldTree = new DirCacheIterator(db.readDirCache());
+		FileTreeIterator newTree = new FileTreeIterator(db);
+
+		//testing a delta filter with one regex (ANY)
+		Pattern deltaFilterPattern = Pattern.compile("change");
+		dfmt.format(dfmt.scan(oldTree, newTree), deltaFilterPattern);
+		dfmt.flush();
+
+		String actual = os.toString("UTF-8");
+		String expected = "";
+
+		assertEquals(expected, actual);
```

Comment:
Combine these - assertEquals("", os.toString("UTF-8"));

Reply:
changed it